src/lib.rs

```diff
@@ -200,7 +282,11 @@ impl PrometheusMetrics {
     }
 
     fn matches(&self, path: &str, method: &Method) -> bool {
-        self.endpoint == path && method == Method::GET
+        if self.endpoint.is_some() {
```

Comment:
I'm personally more inclined to `self.endpoint.is_some() && self.endpoint.as_ref().unwrap() == path && method == Method::GET` _but_ I don't know if compiler optimisations could swap the order of the function calls and make this not equivalent to what you wrote. Not too fussed to keep as is due to that.

Reply:
I'm honestly not sure about this either

I could've done it like this, but [path may be an empty string](https://docs.rs/actix-web/1.0.5/actix_web/http/uri/struct.Uri.html#method.path)
```
self.endpoint.unwrap_or("") == path && method == Method::GET
```